### DISCUSSION TITLE
spatial queries on geopackage feature class

### DISCUSSION THREAD

**Discussion by Eeshdaya** [ngageoint/geopackage-android#69] [Q&A] [answered]
Hello, 
I am trying to query st_area and st_geomfromtext queries that are essential to my app but I keep getting "function not found error" as follows - 

 E/SQLiteLog: (1) no such function: st_area

W/System.err: android.database.sqlite.SQLiteException: no such function: st_area (code 1 SQLITE_ERROR): , while compiling: SELECT "fid", "dimension",  "name", "the_geom" FROM "grids" WHERE st_area(grids.the_geom) > 121;

W/System.err:     at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)

Can anyone point me as to how should I perform spatial functions?

**Comment by bosborn** [comment, marked as answer]
GeoPackages uses basic SQLite (not [SpatialLite](https://www.geopackage.org/#faq)), so the st_area method is not available.

GeoPackage supports a [RTree Spatial Index](https://www.geopackage.org/spec130/#extension_rtree) extension (see [SQLite RTree](https://www.sqlite.org/rtree.html)) to help with geometry range queries.

You'll find some geometry utility methods for already in memory geometries [in this included source](https://github.com/ngageoint/simple-features-java/tree/master/src/main/java/mil/nga/sf/util).

If you really need to query from the area sizes, you could add an area column and save that info with each geometry.
